### PR TITLE
chore(web): refactor publishModal

### PR DESCRIPTION
**Pull request by m-abe-dev**
# Overview

## What I've done
スクリーンショット 2024-08-06 22 07 43

## Summary by CodeRabbit

- **New Features**
	- Enhanced `PublishModal` with improved UI components, including new `Typography` and `ModalPanel`.
	- Introduced a `SwitchField` for toggling search engine indexing options.

- **Improvements**
	- Updated button elements for a clearer presentation experience.
	- Adjusted modal visibility and size properties for better user experience.
	- Enhanced layout with additional padding and consistent text styling.
	- Simplified `PublishToolsPanel` for a more direct button interaction experience.
	- Improved project management functionality by differentiating between story and project contexts.

### CODE REVIEW

File: web/src/beta/features/Editor/Publish/PublishToolsPanel/PublishModal/index.tsx [hunk 1]
```diff
@@ -242,13 +258,14 @@ export default PublishModal;
...
   flex-direction: column;
+  padding: 16px;
   gap: 8px;
   margin-bottom: ${`${spacingSizes["normal"]}px`};
   opacity: ${({ disabled }) => disabled && "0.6"};
   cursor: ${({ disabled }) => disabled && "not-allowed"};
 `;
 
-const Subtitle = styled(Text)`
+const Subtitle = styled(Typography)`
```
Comment (mkumbobeaty) [inline web/src/beta/features/Editor/Publish/PublishToolsPanel/PublishModal/index.tsx:268]: we can not override components style. define wrapper and wrap the component if you want to add some styles 
```suggestion
const SubtitleWrapper = styled("div")`
```

File: web/src/beta/features/Editor/Publish/PublishToolsPanel/PublishModal/index.tsx [hunk 2]
```diff
@@ -242,13 +258,14 @@ export default PublishModal;
 const Section = styled.div<{ disabled?: boolean }>`
```
Comment (mkumbobeaty) [inline web/src/beta/features/Editor/Publish/PublishToolsPanel/PublishModal/index.tsx:258]: use common style format
```suggestion
const Section = styled("div")<{ disabled?: boolean }>(({ disabled, theme }) => ({
  display: "flex",
  padding: theme.spacing.super,
  gap: theme.spacing.small,
  flexDirection: "column",
  marginBottom: `${spacingSizes["normal"]}px`,
  opacity: disabled ? 0.6 : 1,
  cursor: disabled ? "not-allowed" : "auto",
}));`
```

File: web/src/beta/features/Editor/Publish/PublishToolsPanel/PublishModal/index.tsx [hunk 3]
```diff
@@ -266,6 +283,7 @@ const OptionsToggle = styled.div`
   display: flex;
   justify-content: space-between;
   align-items: center;
+  padding: 8px 16px;
```
Comment (mkumbobeaty) [inline web/src/beta/features/Editor/Publish/PublishToolsPanel/PublishModal/index.tsx:286]:
```suggestion
  padding: `${theme.spacing.small}px ${theme.spacing.large}px`,
```

File: web/src/beta/features/Editor/Publish/PublishToolsPanel/PublishModal/index.tsx [hunk 4]
```diff
@@ -274,14 +292,14 @@ const OptionsToggle = styled.div`
...
-    open ? "translateY(10%) rotate(90deg)" : "translateY(0) rotate(180deg)"};
+    open ? "translateY(10%) rotate(270deg)" : "translateY(0) rotate(0deg)"};
 `;
 
 const HideableSection = styled(Section)<{ showOptions?: boolean }>`
   display: ${props => (props.showOptions ? null : "none")};
 `;
 
-const DomainText = styled(Text)`
+const DomainText = styled.div`
```
Comment (mkumbobeaty) [inline web/src/beta/features/Editor/Publish/PublishToolsPanel/PublishModal/index.tsx:302]: consider the common style format

**Review by mkumbobeaty — APPROVED** [review]

**Review by airslice — APPROVED** [review]

**Review by mkumbobeaty — APPROVED** [review]

**Pull request merged**